Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -3,435 +3,408 @@
-    [searchByPhoneNumberButton autoHCenterInSuperview];
-    [searchByPhoneNumberButton autoPinEdge:ALEdgeTop
-                                    toEdge:ALEdgeBottom
-                                    ofView:inviteButton
-                                withOffset:20];
-    [searchByPhoneNumberButton addTarget:self
-                                  action:@selector(hideBackgroundView)
-                        forControlEvents:UIControlEventTouchUpInside];
-}
-
-- (void)hideBackgroundView {
-    [[Environment preferences] setHasDeclinedNoContactsView:YES];
-    
-    [self showEmptyBackgroundViewIfNecessary];
-}
-
-- (void)presentInviteFlow
+- (void)updateTableContents
 {
-    OWSInviteFlow *inviteFlow =
-        [[OWSInviteFlow alloc] initWithPresentingViewController:self contactsManager:self.contactsManager];
-    [self presentViewController:inviteFlow.actionSheetController animated:YES completion:nil];
-}
+    OWSTableContents *contents = [OWSTableContents new];
 
-- (void)showLoadingBackgroundView:(BOOL)show {
-    if (show) {
-        self.searchController.searchBar.hidden = YES;
-        self.tableView.backgroundView          = _loadingBackgroundView;
-        self.tableView.backgroundView.opaque   = YES;
-    } else {
-        self.searchController.searchBar.hidden = NO;
-        self.tableView.backgroundView          = nil;
+    if (self.isNoContactsModeActive) {
+        self.tableViewController.contents = contents;
+        return;
     }
-}
 
-- (void)showEmptyBackgroundViewIfNecessary {
-    self.isNoContactsViewVisible = ([self.contacts count] == 0 &&
-                                    ![[Environment preferences] hasDeclinedNoContactsView]);
-}
+    __weak MessageComposeTableViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
 
-- (void)setIsNoContactsViewVisible:(BOOL)isNoContactsViewVisible {
-    if (isNoContactsViewVisible == _isNoContactsViewVisible) {
-        return;
-    }
-    
-    _isNoContactsViewVisible = isNoContactsViewVisible;
-
-    if (isNoContactsViewVisible) {
-        self.searchController.searchBar.hidden = YES;
-        self.tableView.backgroundView = self.noSignalContactsView;
-        self.tableView.backgroundView.opaque   = YES;
-        self.navigationItem.rightBarButtonItem = nil;
-    } else {
-        self.searchController.searchBar.hidden = NO;
-        self.tableView.backgroundView          = nil;
-        self.navigationItem.rightBarButtonItem = self.createGroupBarButtonItem;
-    }
+    // Group Members
 
-    [self.tableView reloadData];
-}
+    OWSTableSection *section = [OWSTableSection new];
 
-#pragma mark - Initializers
+    [section addItem:[OWSTableItem itemWithCustomCellBlock:^{
+        UITableViewCell *cell = [UITableViewCell new];
+        cell.textLabel.text = NSLocalizedString(
+            @"NEW_CONVERSATION_FIND_BY_PHONE_NUMBER", @"A label the cell that lets you add a new member to a group.");
```

Comment:
Yeah, I'm trying to reuse strings but also avoid having string comments conflict.